### PR TITLE
Add deprecation notes to Syscheck and Syscollector related sections

### PR DESCRIPTION
**Pull request by fdalmaup**
## Description
Part of https://github.com/wazuh/wazuh/issues/20792. Adds deprecation marks to sections related to the `syscheck` and `syscollector` endpoints.

### Docs building
- [x] Compiles without warnings.
### Code formatting and web optimization
- [x] Uses three spaces indentation.
- [x] Adds or updates meta descriptions accordingly.
- [x] Updates the `redirects.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).
### Writing style
- [x] Uses present tense, active voice, and semi-formal registry.
- [x] Uses short, simple sentences.
- [x] Uses **bold** for user interface elements, _italics_ for key terms or emphasis, and `code` font for Bash commands, file names, REST paths, and code.

### CODE REVIEW

**Review by EduLeon12 — DISMISSED** [review]
LGTM!

**Review by Selutario — CHANGES_REQUESTED** [review]
There are two places where `GET /syscollector` is still being mentioned and should add a  deprecation note (and probably updated with the new recommended method, although this will be a core task):
https://github.com/wazuh/wazuh-documentation/blob/c1e5637f11656a597e231404065f8fff06e4ec6d/source/user-manual/capabilities/system-inventory/viewing-system-inventory-data.rst?plain=1#L37-L206

https://github.com/wazuh/wazuh-documentation/blob/c1e5637f11656a597e231404065f8fff06e4ec6d/source/user-manual/capabilities/vulnerability-detection/cpe-helper.rst?plain=1#L166-L596

**Review by Selutario — APPROVED** [review]

File: source/development/wazuh-cluster.rst [hunk 1]
```diff
@@ -481,7 +481,7 @@ Another example that can show how asynchronous tasks are used is Distributed API
 
 The type association with every endpoint can be found here: `API controllers <https://github.com/wazuh/wazuh/tree/v|WAZUH_CURRENT|/api/api/controllers>`__.
 
-Imagine a cluster with two nodes, where there is an agent reporting to the worker node with id *020*. The following diagram shows the process of requesting ``GET/syscollector/020/os`` API endpoint:
+Imagine a cluster with two nodes, where there is an agent reporting to the worker node with id *020*. The following diagram shows the process of requesting ``GET/sca/020`` API endpoint:
```
Comment (javimed) [inline source/development/wazuh-cluster.rst:484]:
```suggestion
Imagine a cluster with two nodes, where there is an agent reporting to the worker node with id *020*. The following diagram shows the process of running a distributed API request by requesting the ``GET/sca/020`` API endpoint:
```

File: source/user-manual/api/rbac/reference.rst [hunk 2]
```diff
@@ -628,11 +628,15 @@ Syscheck
 ^^^^^^^^^^^^^^^
 syscheck:clear
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. deprecated:: 4.9
```
Comment (javimed) [inline source/user-manual/api/rbac/reference.rst:631]:
```suggestion

.. deprecated:: 4.9
```

File: source/user-manual/api/rbac/reference.rst [hunk 3]
```diff
@@ -628,11 +628,15 @@ Syscheck
...
+
 - :api-ref:`DELETE /experimental/syscheck <operation/api.controllers.experimental_controller.clear_syscheck_database>` (`agent:id`_, `agent:group`_)
 - :api-ref:`DELETE /syscheck/{agent_id} <operation/api.controllers.syscheck_controller.delete_syscheck_agent>` (`agent:id`_, `agent:group`_)
 
 syscheck:read
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. deprecated:: 4.9
```
Comment (javimed) [inline source/user-manual/api/rbac/reference.rst:638]:
```suggestion

.. deprecated:: 4.9
```

File: source/user-manual/api/rbac/reference.rst [hunk 4]
```diff
@@ -645,6 +649,8 @@ Syscollector
 ^^^^^^^^^^^^^^^
 syscollector:read
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. deprecated:: 4.9
```
Comment (javimed) [inline source/user-manual/api/rbac/reference.rst:652]:
```suggestion

.. deprecated:: 4.9
```

**Review by javimed — CHANGES_REQUESTED** [review]

**Review by javimed — APPROVED** [review]

**Pull request merged**